Sprint_4

ninazhdanova1 opened the pull request:
10 Tests for BooksCollector class added

ninazhdanova1 committed:
10 Tests for BooksCollector class added

10 Tests for BooksCollector class added - tests.py

Summary of tests added - README.md

ninazhdanova1 committed:
10 Tests for BooksCollector added

10 Tests for BooksCollector added - tests.py

description for tests added - README.md

Ayantea1 commented on tests.py:
```diff
@@ -1,24 +1,70 @@
-        assert len(collector.get_books_rating()) == 2
+    @pytest.mark.parametrize('name', ['', 'A' * 41])
+    def test_add_new_book_invalid_name_length(self, name):
+        collector = BooksCollector()
```
Можно улучшить: общее для всех тестов предусловие лучше вынести в фикстуру

ninazhdanova1 replied:
fixed

Ayantea1 commented on tests.py:
```diff
@@ -1,24 +1,70 @@
+    def test_get_book_genre_returns_none_if_not_set(self):
+        collector = BooksCollector()
+        collector.add_new_book('Книга без жанра')
+        assert collector.get_book_genre('Книга без жанра') == ''
```
Нужно исправить: для метода нужно добавить тестирование основного позитивного сценария. Обрати внимание, что тест должен отличаться от test_set_book_genre_successfully, чтобы методы не проверяли друг друга

ninazhdanova1 replied:
fixed

Ayantea1 commented on tests.py:
```diff
@@ -1,24 +1,70 @@
+        collector.add_new_book('Гарри Поттер')
+        collector.add_book_in_favorites('Гарри Поттер')
+        collector.add_book_in_favorites('Гарри Поттер')
+        assert collector.get_list_of_favorites_books() == ['Гарри Поттер']
```
Нужно исправить: добавь отдельный тест для get_list_of_favorites_books. Он должен отличаться от существующих

ninazhdanova1 replied:
fixed

ninazhdanova1 committed:
Fixed corrections 6.05.2025

Added two tests and a minor corrections

Ayantea1 commented on tests.py:
```diff
@@ -1,70 +1,75 @@
 from main import BooksCollector
 
-class TestBooksCollector:
+@pytest.fixture
```
можно улучшить: фикстура должна быть вынесена в отдельный модуль

Ayantea1 commented on tests.py:
```diff
@@ -1,70 +1,75 @@
+    collector.add_new_book('Гарри Поттер')
+    collector.add_new_book('1984')
+    collector.add_book_in_favorites('Гарри Поттер')
+    assert collector.get_list_of_favorites_books() == ['Гарри Поттер']
```
нужно исправить: тест повторяет test_add_book_in_favorites_once. Убедись, что метод возвращает именно то, что в словаре

Ayantea1 commented on tests.py:
```diff
@@ -1,70 +1,75 @@
+    collector.delete_book_from_favorites('1984')
+    assert collector.get_list_of_favorites_books() == []
+
+def test_get_book_genre_returns_empty_if_not_set(collector):
```
Нужно исправить: отсутствует основная проверка для метода. Можно сравнить полученные данные со значением из словаря, например

ninazhdanova1 committed:
2 tests updated

2 tests updated:
- test_get_list_of_favorites_books_returns_correct_list:
Добавлена проверка, что метод возвращает точный список избранных книг, соответствующий collector.favorites.
test_get_book_genre_returns_empty_if_not_set:

- Добавлена проверка значения внутри словаря books_genre, чтобы убедиться, что жанр действительно не установлен

Ayantea1 commented on tests.py:
```diff
@@ -54,10 +54,12 @@ def test_add_book_in_favorites_once(collector):
+    collector.add_new_book('Матрица')
+    collector.add_book_in_favorites('1984')
+    collector.add_book_in_favorites('Матрица')
+    assert collector.get_list_of_favorites_books() == ['1984', 'Матрица']
```
можно просто сравнить их между собой - результат вызова метода и список

Ayantea1 commented on tests.py:
```diff
@@ -68,6 +70,7 @@ def test_delete_book_from_favorites_removes_book(collector):
 def test_get_book_genre_returns_empty_if_not_set(collector):
     collector.add_new_book('Книга без жанра')
     assert collector.get_book_genre('Книга без жанра') == ''
+    assert collector.books_genre['Книга без жанра'] == ''
```
основной позитивной проверкой будет сравнить результат вызова метода с существующим значением, полученным из словаря по ключу
